src/pinned_memory_manager.cc

```diff
@@ -168,7 +199,7 @@ PinnedMemoryManager::AllocInternal(
   if ((!status.IsOk()) && (*ptr != nullptr)) {
     if (is_pinned) {
       std::lock_guard<std::mutex> lk(pinned_memory_buffer->buffer_mtx_);
```

Comment:
Move these mutex locks within Allocate and Deallocate function.

Reply:
Done 👍